### ISSUE TITLE
False positive: "possibly-unbound-attribute" when accessing attribute after if obj is not None guard

### ISSUE THREAD

**Issue by romainbourdain** [astral-sh/ty#347]
### Summary

Hi,

I'm encountering what seems to be a false positive from Ty when accessing an attribute on an optional object that has just been checked with if.

## Reproductible example
```python
from typing import Optional

class Node:
    def __init__(self, name: str, parent: Optional["Node"] = None):
        self.name = name
        self.parent = parent

positions = [("root", (0, 0)), ("child", (1, 1))]

node = Node("child", parent=Node("root"))

if node.parent:
    pos = next(p for name, p in positions if name == node.parent.name)[1]
```

## Ty output
```
Attribute `name` on type `Unknown | Node | None` is possibly unboundty(possibly-unbound-attribute)
```

**Comment by sharkdp** [comment]
Thank you for reporting this. We currently don't support narrowing on attribute expressions, yet. See #164.
